### ISSUE TITLE
Transient check python failure.

### ISSUE THREAD

**Issue by davinchia** [xvik/gradle-use-python-plugin#19]
Hi!

Thanks for the great plugin. Currently using it in an open-source project and loving it.

We are running into a transient error with check python task. 

This is the stacktrace:
```
 * Exception is:
2021-05-11T13:17:25.3774839Z 2021-05-11T13:17:23.588+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':airbyte-integrations:bases:airbyte-protocol:checkPython'.
2021-05-11T13:17:25.3779184Z 2021-05-11T13:17:23.588+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:200)
2021-05-11T13:17:25.3782990Z 2021-05-11T13:17:23.588+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:263)
2021-05-11T13:17:25.3787163Z 2021-05-11T13:17:23.589+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:198)
2021-05-11T13:17:25.3792255Z 2021-05-11T13:17:23.589+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:179)
2021-05-11T13:17:25.3797174Z 2021-05-11T13:17:23.589+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:109)
2021-05-11T13:17:25.3802275Z 2021-05-11T13:17:23.589+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
2021-05-11T13:17:25.3807682Z 2021-05-11T13:17:23.589+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
2021-05-11T13:17:25.3813055Z 2021-05-11T13:17:23.589+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
2021-05-11T13:17:25.3817862Z 2021-05-11T13:17:23.589+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
2021-05-11T13:17:25.3822544Z 2021-05-11T13:17:23.589+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
2021-05-11T13:17:25.3827239Z 2021-05-11T13:17:23.589+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
2021-05-11T13:17:25.3831491Z 2021-05-11T13:17:23.589+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
2021-05-11T13:17:25.3835707Z 2021-05-11T13:17:23.589+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
2021-05-11T13:17:25.3840249Z 2021-05-11T13:17:23.589+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
2021-05-11T13:17:25.3845088Z 2021-05-11T13:17:23.589+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
2021-05-11T13:17:25.3849629Z 2021-05-11T13:17:23.589+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
2021-05-11T13:17:25.3853804Z 2021-05-11T13:17:23.589+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
2021-05-11T13:17:25.3858118Z 2021-05-11T13:17:23.589+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
2021-05-11T13:17:25.3862618Z 2021-05-11T13:17:23.590+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
2021-05-11T13:17:25.3867169Z 2021-05-11T13:17:23.590+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
2021-05-11T13:17:25.3871704Z 2021-05-11T13:17:23.590+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:76)
2021-05-11T13:17:25.3876967Z 2021-05-11T13:17:23.590+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
2021-05-11T13:17:25.3882105Z 2021-05-11T13:17:23.590+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:76)
2021-05-11T13:17:25.3886731Z 2021-05-11T13:17:23.590+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
2021-05-11T13:17:25.3890909Z 2021-05-11T13:17:23.590+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
2021-05-11T13:17:25.3895128Z 2021-05-11T13:17:23.590+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:372)
2021-05-11T13:17:25.3899694Z 2021-05-11T13:17:23.590+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:359)
2021-05-11T13:17:25.3904554Z 2021-05-11T13:17:23.590+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:352)
2021-05-11T13:17:25.3909476Z 2021-05-11T13:17:23.590+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:338)
2021-05-11T13:17:25.3913782Z 2021-05-11T13:17:23.590+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
2021-05-11T13:17:25.3917419Z 2021-05-11T13:17:23.590+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
2021-05-11T13:17:25.3921169Z 2021-05-11T13:17:23.590+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
2021-05-11T13:17:25.3924924Z 2021-05-11T13:17:23.590+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
2021-05-11T13:17:25.3928598Z 2021-05-11T13:17:23.590+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
2021-05-11T13:17:25.3932204Z 2021-05-11T13:17:23.590+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
2021-05-11T13:17:25.3935851Z 2021-05-11T13:17:23.590+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
2021-05-11T13:17:25.3939720Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Caused by: ru.vyarus.gradle.plugin.python.util.PythonExecutionFailed: Python call failed: python3 -m pip install virtualenv==20.4.2 --user
2021-05-11T13:17:25.3943714Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
2021-05-11T13:17:25.3947899Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
2021-05-11T13:17:25.3952509Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
2021-05-11T13:17:25.3956655Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at ru.vyarus.gradle.plugin.python.cmd.Python.processExecution(Python.groovy:403)
2021-05-11T13:17:25.3960044Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at ru.vyarus.gradle.plugin.python.cmd.Python.access$2(Python.groovy)
2021-05-11T13:17:25.3963195Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at ru.vyarus.gradle.plugin.python.cmd.Python$_exec_closure7.doCall(Python.groovy:249)
2021-05-11T13:17:25.3966282Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at ru.vyarus.gradle.plugin.python.cmd.Python$_exec_closure7.call(Python.groovy)
2021-05-11T13:17:25.3969346Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at ru.vyarus.gradle.plugin.python.cmd.Python.exec(Python.groovy:249)
2021-05-11T13:17:25.3972517Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at ru.vyarus.gradle.plugin.python.cmd.Python.callModule(Python.groovy:260)
2021-05-11T13:17:25.3975624Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at ru.vyarus.gradle.plugin.python.cmd.Pip.exec(Pip.groovy:150)
2021-05-11T13:17:25.3978600Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at ru.vyarus.gradle.plugin.python.cmd.Pip.install(Pip.groovy:112)
2021-05-11T13:17:25.3982088Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at ru.vyarus.gradle.plugin.python.task.CheckPythonTask.checkEnv(CheckPythonTask.groovy:126)
2021-05-11T13:17:25.3985909Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at ru.vyarus.gradle.plugin.python.task.CheckPythonTask.run(CheckPythonTask.groovy:55)
2021-05-11T13:17:25.3989440Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2021-05-11T13:17:25.3993179Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2021-05-11T13:17:25.3997145Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2021-05-11T13:17:25.4000695Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
2021-05-11T13:17:25.4004493Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:58)
2021-05-11T13:17:25.4008917Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:51)
2021-05-11T13:17:25.4013382Z 2021-05-11T13:17:23.591+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:29)
2021-05-11T13:17:25.4017824Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.run(ExecuteActionsTaskExecuter.java:555)
2021-05-11T13:17:25.4022133Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
2021-05-11T13:17:25.4026391Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
2021-05-11T13:17:25.4030594Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
2021-05-11T13:17:25.4034761Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
2021-05-11T13:17:25.4039076Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
2021-05-11T13:17:25.4043587Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
2021-05-11T13:17:25.4047995Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:56)
2021-05-11T13:17:25.4052441Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$run$1(DefaultBuildOperationExecutor.java:71)
2021-05-11T13:17:25.4057631Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.runWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:45)
2021-05-11T13:17:25.4062697Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:71)
2021-05-11T13:17:25.4067662Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:540)
2021-05-11T13:17:25.4073013Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:523)
2021-05-11T13:17:25.4078057Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.access$300(ExecuteActionsTaskExecuter.java:108)
2021-05-11T13:17:25.4083236Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.executeWithPreviousOutputFiles(ExecuteActionsTaskExecuter.java:271)
2021-05-11T13:17:25.4088295Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.execute(ExecuteActionsTaskExecuter.java:260)
2021-05-11T13:17:25.4092386Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.ExecuteStep.lambda$execute$1(ExecuteStep.java:34)
2021-05-11T13:17:25.4095942Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:34)
2021-05-11T13:17:25.4099475Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:26)
2021-05-11T13:17:25.4103358Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:67)
2021-05-11T13:17:25.4107445Z 2021-05-11T13:17:23.592+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:36)
2021-05-11T13:17:25.4111689Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:49)
2021-05-11T13:17:25.4116184Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:34)
2021-05-11T13:17:25.4120448Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:43)
2021-05-11T13:17:25.4124613Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:73)
2021-05-11T13:17:25.4128507Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:54)
2021-05-11T13:17:25.4132279Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:44)
2021-05-11T13:17:25.4136303Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:54)
2021-05-11T13:17:25.4140378Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:38)
2021-05-11T13:17:25.4144939Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:42)
2021-05-11T13:17:25.4149452Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.CacheStep.executeWithoutCache(CacheStep.java:159)
2021-05-11T13:17:25.4153144Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:72)
2021-05-11T13:17:25.4156575Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:43)
2021-05-11T13:17:25.4160487Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:44)
2021-05-11T13:17:25.4164893Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:33)
2021-05-11T13:17:25.4169071Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:38)
2021-05-11T13:17:25.4182267Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:24)
2021-05-11T13:17:25.4186568Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:92)
2021-05-11T13:17:25.4190669Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$0(SkipUpToDateStep.java:85)
2021-05-11T13:17:25.4194712Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:55)
2021-05-11T13:17:25.4198640Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:39)
2021-05-11T13:17:25.4202607Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:76)
2021-05-11T13:17:25.4206641Z 2021-05-11T13:17:23.593+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:37)
2021-05-11T13:17:25.4211621Z 2021-05-11T13:17:23.594+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:36)
2021-05-11T13:17:25.4217531Z 2021-05-11T13:17:23.594+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:26)
2021-05-11T13:17:25.4222656Z 2021-05-11T13:17:23.594+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:94)
2021-05-11T13:17:25.4227166Z 2021-05-11T13:17:23.594+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:49)
2021-05-11T13:17:25.4231966Z 2021-05-11T13:17:23.594+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:79)
2021-05-11T13:17:25.4237114Z 2021-05-11T13:17:23.594+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:53)
2021-05-11T13:17:25.4241525Z 2021-05-11T13:17:23.594+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:74)
2021-05-11T13:17:25.4245315Z 2021-05-11T13:17:23.594+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.lambda$execute$2(SkipEmptyWorkStep.java:78)
2021-05-11T13:17:25.4249261Z 2021-05-11T13:17:23.594+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:78)
2021-05-11T13:17:25.4253232Z 2021-05-11T13:17:23.594+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:34)
2021-05-11T13:17:25.4258098Z 2021-05-11T13:17:23.594+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:39)
2021-05-11T13:17:25.4263111Z 2021-05-11T13:17:23.594+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:40)
2021-05-11T13:17:25.4267559Z 2021-05-11T13:17:23.594+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:28)
2021-05-11T13:17:25.4271732Z 2021-05-11T13:17:23.594+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.impl.DefaultWorkExecutor.execute(DefaultWorkExecutor.java:33)
2021-05-11T13:17:25.4276484Z 2021-05-11T13:17:23.594+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:187)
2021-05-11T13:17:25.4281566Z 2021-05-11T13:17:23.594+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:179)
2021-05-11T13:17:25.4286480Z 2021-05-11T13:17:23.594+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:109)
2021-05-11T13:17:25.4291568Z 2021-05-11T13:17:23.594+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
2021-05-11T13:17:25.4296902Z 2021-05-11T13:17:23.594+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
2021-05-11T13:17:25.4302181Z 2021-05-11T13:17:23.595+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
2021-05-11T13:17:25.4307078Z 2021-05-11T13:17:23.595+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
2021-05-11T13:17:25.4311765Z 2021-05-11T13:17:23.595+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
2021-05-11T13:17:25.4316454Z 2021-05-11T13:17:23.595+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
2021-05-11T13:17:25.4320779Z 2021-05-11T13:17:23.595+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
2021-05-11T13:17:25.4324960Z 2021-05-11T13:17:23.595+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
2021-05-11T13:17:25.4329508Z 2021-05-11T13:17:23.595+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
2021-05-11T13:17:25.4334313Z 2021-05-11T13:17:23.595+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
2021-05-11T13:17:25.4338786Z 2021-05-11T13:17:23.595+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
2021-05-11T13:17:25.4342959Z 2021-05-11T13:17:23.595+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
2021-05-11T13:17:25.4347400Z 2021-05-11T13:17:23.595+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
2021-05-11T13:17:25.4351890Z 2021-05-11T13:17:23.595+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
2021-05-11T13:17:25.4356344Z 2021-05-11T13:17:23.595+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
2021-05-11T13:17:25.4360869Z 2021-05-11T13:17:23.595+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:76)
2021-05-11T13:17:25.4366088Z 2021-05-11T13:17:23.595+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
2021-05-11T13:17:25.4371270Z 2021-05-11T13:17:23.595+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:76)
2021-05-11T13:17:25.4375822Z 2021-05-11T13:17:23.595+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
2021-05-11T13:17:25.4379998Z 2021-05-11T13:17:23.595+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
2021-05-11T13:17:25.4384307Z 2021-05-11T13:17:23.595+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:372)
2021-05-11T13:17:25.4388875Z 2021-05-11T13:17:23.596+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:359)
2021-05-11T13:17:25.4393618Z 2021-05-11T13:17:23.596+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:352)
2021-05-11T13:17:25.4398585Z 2021-05-11T13:17:23.596+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:338)
2021-05-11T13:17:25.4402808Z 2021-05-11T13:17:23.596+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
2021-05-11T13:17:25.4406450Z 2021-05-11T13:17:23.596+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
2021-05-11T13:17:25.4410210Z 2021-05-11T13:17:23.596+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
2021-05-11T13:17:25.4414004Z 2021-05-11T13:17:23.596+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
2021-05-11T13:17:25.4417662Z 2021-05-11T13:17:23.596+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
2021-05-11T13:17:25.4421319Z 2021-05-11T13:17:23.596+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
2021-05-11T13:17:25.4425063Z 2021-05-11T13:17:23.596+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
2021-05-11T13:17:25.4427800Z 2021-05-11T13:17:23.596+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2021-05-11T13:17:25.4429785Z 2021-05-11T13:17:23.596+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2021-05-11T13:17:25.4431990Z 2021-05-11T13:17:23.596+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Get more help at https://help.gradle.org
```

This is on 2.3.0. We are running this as part of main build on a [Github workflow](https://github.com/airbytehq/airbyte/blob/master/.github/workflows/gradle.yml#L102) as part of a [plugin](https://github.com/airbytehq/airbyte/blob/master/buildSrc/src/main/groovy/airbyte-python.gradle#L15) we've written.

I can't seem to find a pattern to when this happens.

Any ideas why this is happening?

Thanks!

**Comment by xvik** [comment]
Hi!
You need look at the python output just before the exception. There should be an actual python error.

It's impossible to tell anything with just an exception (it's used just to fail build), but my guess is that pip version is the problem (conflicts with virtualenv).
If this is the case then you'll need to update pip manually (`python3 -m pip install --upgrade pip`) and only after that run gradle.
I noted that you specify pip 21 in the gradle build, but it would be installed only inside virtualenv.

Alternatively, you can try to downgrade the installed virtualenv (as you use python 3.7 it should fit better):
```
python {
    virtualenvVersion = '16.7.9'
}
```
(virtualenv 20 could conflict with the old pip version often coming with linux distros... I already faced such cases)

**Comment by davinchia** [comment]
I see. Might this be it:
```
2021-05-11T13:17:25.3679131Z 2021-05-11T13:17:07.452+0000 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Build operation 'Execute Project.afterEvaluate listener' started
2021-05-11T13:17:25.3681890Z 2021-05-11T13:17:07.452+0000 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Completing Build operation 'Execute Project.afterEvaluate listener'
2021-05-11T13:17:25.3685030Z 2021-05-11T13:17:07.452+0000 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Completing Build operation 'Notify afterEvaluate listeners of :airbyte-integrations:connectors:destination-meilisearch'
2021-05-11T13:17:25.3738634Z 2021-05-11T13:17:12.358+0000 [ERROR] [org.gradle.api.Project] 	 WARNING: Package(s) not found: virtualenv
2021-05-11T13:17:25.3739856Z 2021-05-11T13:17:13.008+0000 [ERROR] [org.gradle.api.Project] 	 WARNING: Package(s) not found: virtualenv
2021-05-11T13:17:25.3741053Z 2021-05-11T13:17:12.885+0000 [ERROR] [org.gradle.api.Project] 	 WARNING: Package(s) not found: virtualenv
2021-05-11T13:17:25.3742368Z 2021-05-11T13:17:13.112+0000 [ERROR] [org.gradle.api.Project] 	 WARNING: Package(s) not found: virtualenv
2021-05-11T13:17:25.3743684Z 2021-05-11T13:17:13.026+0000 [ERROR] [org.gradle.api.Project] 	 WARNING: Package(s) not found: virtualenv
2021-05-11T13:17:25.3744870Z 2021-05-11T13:17:14.092+0000 [ERROR] [org.gradle.api.Project] 	 WARNING: Package(s) not found: virtualenv
2021-05-11T13:17:25.3746571Z 2021-05-11T13:17:19.321+0000 [ERROR] [system.err] Note: /actions-runner/_work/airbyte/airbyte/airbyte-commons/src/main/java/io/airbyte/commons/util/MoreIterators.java uses unchecked or unsafe operations.
2021-05-11T13:17:25.3748026Z 2021-05-11T13:17:19.322+0000 [ERROR] [system.err] Note: Recompile with -Xlint:unchecked for details.
```

**Comment by xvik** [comment]
> WARNING: Package(s) not found: virtualenv

Yes, looks like your problem. But have no idea how this could happen.
I have not seen errors like this before and not sure how this could be even possible (maybe python installation is concurrent and something is not finished sometimes (never used actions myself)). 

I can only suggest installing virtualenv manually with direct command before running gradle:
```
python3 -m pip install virtualenv
```

* maybe in global scope would work better than `--user` scope; 
* also, it might be better to install a lower virtualenv version: `virtualenv<20`

If you would succeed in manual installation, the plugin would simply use globally installed virtualenv for local environment creation.

At least this way problem should be localized to python itself instead of gradle.

**Comment by xvik** [comment]
I saw the linked issue only by accident (github didn't notify about it). 

The problem caused by [parallel tasks execution](https://github.com/airbytehq/airbyte/blob/961ab6693ad4dcb4fa5cdf2a77be42adbf0cf6c9/gradle.properties#L1)


```
> Task :airbyte-integrations:connectors:source-file:checkPython FAILED
	 WARNING: Package(s) not found: virtualenv
Using python 3.7.10 from /opt/hostedtoolcache/Python/3.7.10/x64 (python3)
Using pip 21.1.2 from /opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pip (python 3.7)
[python] python3 -m pip install virtualenv==16.7.9 --user
	 Collecting virtualenv==16.7.9
	   Downloading virtualenv-16.7.9-py2.py3-none-any.whl (3.4 MB)
	 Installing collected packages: virtualenv
	 ERROR: Could not install packages due to an OSError: [Errno 17] File exists: '/home/runner/.local'
	 
	 WARNING: Package(s) not found: virtualenv

> Task :airbyte-integrations:bases:airbyte-protocol:checkPython
Using python 3.7.10 from /opt/hostedtoolcache/Python/3.7.10/x64 (python3)
Using pip 21.1.2 from /opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pip (python 3.7)
[python] python3 -m pip install virtualenv==16.7.9 --user
	 Collecting virtualenv==16.7.9
	   Downloading virtualenv-16.7.9-py2.py3-none-any.whl (3.4 MB)
	 Installing collected packages: virtualenv
	 Successfully installed virtualenv-16.7.9
```

Both `:airbyte-integrations:connectors:source-file:checkPython` and `:airbyte-integrations:bases:airbyte-protocol:checkPython` installs virtualenv package in parallel: one succeeds and another throwns python error: 

```
ERROR: Could not install packages due to an OSError: [Errno 17] File exists: '/home/runner/.local'
```

To fix this you can do manual virtualenv package installation in github action (before running gradle build): `python3 -m pip install virtualenv==16.7.9 --user` (actually any virtualenv version would be ok). 

I'll see later what I can do with parallel executions in the plugin itself.

**Comment by davinchia** [comment]
@xvik let me give it a shot!

**Comment by davinchia** [comment]
Explicitly installing seemed to have worked - tested on a branch. Have merged into production. Will update!
